### PR TITLE
Measure splitting

### PR DESCRIPTION
**Pull request by jrmsgr**
Completed the measure splitting function for the Staff class. That function finds the staff's bar lines, crops them out, and stores them in the measure_list attribute. 

Notes: 
-a (temporary ?) line_gap attribute has been added to Staff, as this information is needed for the function.
-a complete image preprocessing is included in the function,  it will probably be removed in future versions.
-for now, the function does not distinguish repeat signs **: ||** from the end of the score **||**, as only vertical lines are detected. A solution will have to be found (i did not look into it).

### CODE REVIEW

File: music_detection/staff.py [hunk 1]
```diff
@@ -1,13 +1,62 @@
...
         self.measure_list = []
+        self.image = image #Actual picture of the staff
+        self.line_gap = 0 #Added it just to signify that i need it from a previous processing
+
+    def identify_measures(self) -> None:
+        '''Split the staff into different Measure objects stored in measure_list'''
+        vote_threshold = int(3.25*self.line_gap) #Determined experimentally
+
+        #Preprocessing : will be probably be removed in later versions, because
+        #performed in another function
```
Comment (Ratoone) [inline music_detection/staff.py]: Consider writing TODO in front, and maybe even write a github issue

File: music_detection/staff.py [hunk 2]
```diff
@@ -1,13 +1,62 @@
...
+            temp = cv2.subtract(im_morph, im_open)
+            eroded = cv2.erode(im_morph, element)
+            im_skel = cv2.bitwise_or(im_skel, temp)
+            im_morph = eroded.copy()
+
+        #Extraction of bar lines'x-coordinates
+        lines = cv2.HoughLines(im_skel, 1, np.pi/180, vote_threshold)
+        bar_lines_loc = list()
+        for rho, theta in lines[:,0]:
+            if theta==0: #only vertical lines are kept
```
Comment (Ratoone) [inline music_detection/staff.py]: Maybe the condition is a bit strict, consider taking a small threshold (like 5-10 degrees)

File: music_detection/staff.py [hunk 3]
```diff
@@ -1,13 +1,62 @@
...
+                bar_lines_loc.append(int(round(rho)))
+        
+        bar_lines_loc.sort() #Necessary because bar lines are found in a random order
```
Comment (Ratoone) [inline music_detection/staff.py:51]: It's not random, but I got the idea

File: music_detection/staff.py [hunk 4]
```diff
@@ -1,13 +1,62 @@
...
+            
+            #If the 2 lines are far enough from each other, the measure in-between is simply
+            #added to the list
+            if next_boundary-prev_boundary > 2*self.line_gap: 
+                measure = Measure(self.image[:, prev_boundary+1:next_boundary])#Measure is cropped
+                self.measure_list.append(measure)
+            else: 
+                #Otherwise it means that the previous mesure has to be repeated
+                #WARNING : for now, this algorithm does not differenciate repeating measures (: ||)
+                #and the end of a music score (||)
```
Comment (Ratoone) [inline music_detection/staff.py]: Again, consider writing a TODO in front and/or open an issue

**Review by Ratoone — CHANGES_REQUESTED** [review]
Consider moving the test into the test folder and put it in a test function. Check dummy_test or others from the folder for reference. Looks fine otherwise

File: music_detection/staff.py (hunk 2 above)
Comment (Ratoone) [reply music_detection/staff.py]: Especially considering that the image rectification may not be perfect

Comment (jrmsgr) [reply music_detection/staff.py]: I know it is a bit strict, but I skeletize the image beforehand to make sure that I only keep vertical lines, thus avoiding such a case. I will look into it anyway and see what I can do.

File: test/test_measure_splitting.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import cv2
+import pytest
+from music_detection.staff import Staff
+
+
+@pytest.fixture(scope="session", autouse=True)
```
Comment (Ratoone) [inline test/test_measure_splitting.py]: You don't have a fixture for this test, so you don't really need this

**Review by Ratoone — CHANGES_REQUESTED** [review]
I think you committed your local test folder by mistake (somehow). Also, you should rename the test file to measure_splitting_test, otherwise pytest doesn't find it

**Review by Ratoone — APPROVED** [review]

**Pull request merged**